test/e2e/operator_test.go

```diff
@@ -192,14 +160,14 @@ func TestExternalDNSRecordLifecycle(t *testing.T) {
 	}
 
 	t.Log("Creating credentials secret")
-	resourceSecret := helper.makeCredentialsSecret(testCredSecretName)
+	resourceSecret := helper.makeCredentialsSecret()
 	err = kubeClient.Create(context.TODO(), resourceSecret)
 	if err != nil {
 		t.Fatalf("Failed to create credentials secret %s/%s for resource: %v", resourceSecret.Namespace, resourceSecret.Name, err)
 	}
 
 	t.Log("Creating external dns instance")
-	extDNS := defaultExternalDNS(t, testExtDNSName, hostedZoneID, hostedZoneDomain, resourceSecret, helper.platform(), providerOptions)
+	extDNS := helper.defaultExternalDNS(resourceSecret)
```

Comment:
I agree that passing `t`, `platform` and `providerOptions` doesn't make sense anymore (`hostedZoneDomain` too as it was never used) . However for the rest of the variables, I'd prefer to see them passed as input parameters, for the same reason of the dependency on the global vars.

Reply:
hostedZoneDomain, hostedZoneId both were used, so we have to pass these values to inline with previous review comments.